### PR TITLE
Mark

### PR DESCRIPTION
**Pull request by mreid1130**

### CODE REVIEW

File: ical.js [hunk 1]
```diff
@@ -27,152 +30,178 @@
     )
   }
 
-  var parseParams = function(p){
+  var parseParams = function(p) {
+    console.log('parseParams');
```
Comment (cdelauder) [inline ical.js]: Let's remove these or we will end up with a lot of noise in the logs

File: ical.js [hunk 2]
```diff
@@ -27,152 +30,178 @@
...
+   * @return {object} The Javascript date object
+   */
+  function parseDate(val, params, curr) {
+    var objToReturn = val;
+    if (params && params[0] === "VALUE=DATE") {
+      // Just Date
+
+      var comps = /^(\d{4})(\d{2})(\d{2})$/.exec(val);
+      if (comps !== null) {
+        // No TZ info - assume same timezone as this computer
```
Comment (cdelauder) [inline ical.js]: I don't think this comment is still true.

File: ical.js [hunk 3]
```diff
@@ -192,92 +219,95 @@
...
+      'DTEND': dateParam('end'),
+      ' CLASS': storeParam('class'),
+      'TRANSP': storeParam('transparency'),
+      'GEO': geoParam('geo'),
+      'PERCENT-COMPLETE': storeParam('completion'),
+      'COMPLETED': dateParam('completed'),
+      'CATEGORIES': categoriesParam('categories'),
+      'FREEBUSY': freebusyParam('freebusy'),
+      'EXDATE': dateParamArray('exdate'),
+      'RECURRENCE-ID': storeParam('recurrenceId')
```
Comment (cdelauder) [inline ical.js:270]: do we need an RRULE or recurrence field?

File: ical.js [hunk 4]
```diff
@@ -192,92 +219,95 @@
...
-      , 'FREEBUSY': freebusyParam('freebusy')
+      ,
+      'SUMMARY': storeParam('summary'),
+      'DESCRIPTION': storeParam('description'),
+      'URL': storeParam('url'),
+      'UID': storeParam('uid'),
+      'LOCATION': storeParam('location'),
+      'DTSTART': dateParam('start'),
+      'DTEND': dateParam('end'),
+      ' CLASS': storeParam('class'),
```
Comment (cdelauder) [inline ical.js]: I checked the spec and there is no reason to believe that this has whitespace before it. This is definitely a typo

File: ical.js [hunk 5]
```diff
@@ -27,152 +30,176 @@
...
+      if (comps[7] == 'Z') { // GMT
+        return new Date(Date.UTC(
+          parseInt(comps[1], 10),
+          parseInt(comps[2], 10) - 1,
+          parseInt(comps[3], 10),
+          parseInt(comps[4], 10),
+          parseInt(comps[5], 10),
+          parseInt(comps[6], 10)
+        ));
+        // TODO add tz
```
Comment (cdelauder) [inline ical.js]: remove this comment since you have now added it

File: ical.js (hunk 3 above)
Comment (mreid1130) [inline ical.js:270]: I noticed that too, the data is coming in from the importer correctly, and it looks like handleObject function will create the object correctly, but I'll play around with hard coding `RRULE` into this list tomorrow

**Pull request merged**